src/app/modules/submission/email-submission/email-submission.util.ts

```diff
@@ -550,3 +599,131 @@ export const maskUidOnLastField = (
+}
+
+/**
+ * Function to extract information for email form field from response
+ * Form field is used to send responses to admin
+ */
+const getFormFormattedResponse = (
+  response: ResponseFormattedForEmail,
+  hashedFields: Set<string>,
+): EmailAdminDataField => {
+  const { answer, fieldType } = response
+  const answerSplitByNewLine = answer.split('\n')
+  const formData = {
+    question: getFormDataPrefixedQuestion(response, hashedFields),
+    answerTemplate: answerSplitByNewLine,
+    answer,
+    fieldType,
+  }
+  return formData
+}
+
+/**
+ * Function to extract information for email autoreply field from response
+ * Autoreply field is used to send confirmation emails
+ */
+const getAutoReplyFormattedResponse = (
+  response: ResponseFormattedForEmail,
+): EmailRespondentConfirmationField | undefined => {
+  const { question, answer, isVisible } = response
+  const answerSplitByNewLine = answer.split('\n')
+  // Auto reply email will contain only visible fields
+  if (isVisible) {
+    const autoReplyData = {
+      question, // No prefixes for autoreply
+      answerTemplate: answerSplitByNewLine,
+    }
+    return autoReplyData
+  }
+  return undefined
+}
+
+export class EmailDataObj {
+  parsedResponses: ProcessedFieldResponse[]
+  hashedFields: Set<string>
+  authType: AuthType
+
+  constructor(
+    parsedResponses: ProcessedFieldResponse[],
+    hashedFields: Set<string>,
+    authType: AuthType,
+  ) {
+    this.parsedResponses = parsedResponses
+    this.hashedFields = hashedFields
+    this.authType = authType
+  }
+
+  /**
+   * Getter function to return dataCollationData which is used for data collation tool
+   */
+  get dataCollationData(): EmailDataCollationToolField[] {
+    const dataCollationFormattedData = this.parsedResponses.flatMap(
+      (response) =>
+        createFormattedDataForOneField(
+          response,
+          this.hashedFields,
+          getDataCollationFormattedResponse,
+        ),
+    )
+
+    // Compact is necessary because getDataCollationFormattedResponse
+    // will return undefined for header fields
+    return compact(dataCollationFormattedData)
+  }
+
+  /**
+   * Getter function to return autoReplyData for confirmation emails to respondent
+   * If AuthType is CP, return a masked version
+   */
+  get autoReplyData(): EmailRespondentConfirmationField[] {
+    const unmaskedAutoReplyFormattedData = this.parsedResponses.flatMap(
+      (response) =>
+        createFormattedDataForOneField(
+          response,
+          this.hashedFields,
+          getAutoReplyFormattedResponse,
+        ),
+    )
+
+    // Compact is necessary because getAutoReplyFormattedResponse
+    // will return undefined for non-visible fields
+    const unmaskedAutoReplyData = compact(unmaskedAutoReplyFormattedData)
+    const maskedAutoReplyData = maskUidOnLastField(unmaskedAutoReplyData)
+
+    return this.authType === AuthType.CP
+      ? maskedAutoReplyData
+      : unmaskedAutoReplyData
```

Comment:
Simplifying this by (1) removing the need for `unmaskedAutoReplyFormattedData` so that less mutable state needs to be held, and (2) folding `maskUidOnLastField` into the ternary op so that it isn't run unnecessarily

```suggestion
    // Compact is necessary because getAutoReplyFormattedResponse
    // will return undefined for non-visible fields
    const unmaskedAutoReplyData = compact(
      this.parsedResponses.flatMap(
        (response) =>
          createFormattedDataForOneField(
            response,
            this.hashedFields,
            getAutoReplyFormattedResponse,
          ),
      )
    )

    return this.authType === AuthType.CP
      ? maskUidOnLastField(unmaskedAutoReplyData)
      : unmaskedAutoReplyData
```